tests/gis_tests/geoadmin/tests.py

```diff
@@ -30,7 +30,7 @@ def test_widget_invalid_string(self):
             ' rows="10" name="point" hidden></textarea>',
             output,
         )
-        self.assertEqual(len(cm.records), 1)
+        self.assertEqual(len(cm.records), 2)
```

Comment:
I'm not sure about this change. It seems wrong to me. Investigating...

Reply:
See. https://github.com/django/django/pull/17119

So I don't think this is too different to current behaviour. The value is deserialized once for errors and once to render the widget. Errors are cached so would still be two log entries when rendering a form with this patch.

Reply:
Fine for me.